File: library/src/main/java/com/amar/library/ui/presentation/IStickyScrollPresentation.java
package com.amar.library.ui.presentation;

/**
 * Created by Amar Jain on 17/03/17.
 */

public interface IStickyScrollPresentation {
    void freeHeader();
    void freeFooter();
    void stickHeader(int translationY);
    void stickFooter(int translationY);

    void initHeaderView(int id);
    void initFooterView(int id);

}

File: library/src/main/java/com/amar/library/ui/presentation/IStickyScrollPresentation.kt
package com.amar.library.ui.presentation

interface IStickyScrollPresentation {
    val currentScrollYPos: Int
    fun freeHeader()
    fun freeFooter()
    fun stickHeader(translationY: Int)
    fun stickFooter(translationY: Int)
    fun initHeaderView(id: Int)
    fun initFooterView(id: Int)
}